DOC: update the pandas.DataFrame.pct_change docstring

myles opened the pull request:
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################### Docstring (pandas.DataFrame.pct_change)  ###################
################################################################################

Percentage change between the current and previous element.

This is useful in comparing the percentage of change in a time series
of elements.

Parameters
----------
periods : int, default 1
    Periods to shift for forming percent change.
fill_method : str, default 'pad'
    How to handle NAs before computing percent changes.
limit : int, default None
    The number of consecutive NAs to fill before stopping.
freq : DateOffset, timedelta, or offset alias string, optional
    Increment to use from time series API (e.g. 'M' or BDay()).
**kwargs : mapping, optional
    Additional keyword arguments are passed into
    `DataFrame.shift` or `Series.shift`.

Returns
-------
chg : Series or DataFrame
    The same type as the calling object.

Examples
--------
See the percentage change in a Series.

>>> s = pd.Series([90, 91, 85])
>>> s
0    90
1    91
2    85
dtype: int64
>>> s.pct_change()
0         NaN
1    0.011111
2   -0.065934
dtype: float64

See the percentage change in a Series where filling NAs with last
valid observation forward to next valid.

>>> s = pd.Series([90, 91, None, 85])
>>> s
0    90.0
1    91.0
2     NaN
3    85.0
dtype: float64
>>> s.pct_change(fill_method='ffill')
0         NaN
1    0.011111
2    0.000000
3   -0.065934
dtype: float64

Percentage change in French franc, Deutsche Mark, and Italian lira from
1 January 1980 to 1 March 1980.

>>> df = pd.DataFrame({
...     'FR': [4.0405, 4.0963, 4.3149],
...     'GR': [1.7246, 1.7482, 1.8519],
...     'IT': [804.74, 810.01, 860.13]},
...     index=['1980-01-01', '1980-02-01', '1980-03-01'])
>>> df
                FR      GR      IT
1980-01-01  4.0405  1.7246  804.74
1980-02-01  4.0963  1.7482  810.01
1980-03-01  4.3149  1.8519  860.13
>>> df.pct_change()
                  FR        GR        IT
1980-01-01       NaN       NaN       NaN
1980-02-01  0.013810  0.013684  0.006549
1980-03-01  0.053365  0.059318  0.061876

Percentage of change in GOOG and APPL stock volume.

>>> df = pd.DataFrame({
...     '2016': [1769950, 30586265],
...     '2015': [1500923, 40912316],
...     '2014': [1371819, 41403351]},
...     index=['GOOG', 'APPL'])
>>> df
          2016      2015      2014
GOOG   1769950   1500923   1371819
APPL  30586265  40912316  41403351
>>> df.pct_change(axis='columns')
      2016      2015      2014
GOOG   NaN -0.151997 -0.086016
APPL   NaN  0.337604  0.012002

See Also
--------
DataFrame.diff : see the difference of two columns
Series.diff : see the difference of two columns

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameters {'kwargs'} not documented
		Unknown parameters {'**kwargs'}
```

myles committed:
:bulb: Fix some basic stuff from the validator script.

myles committed:
:bulb: Add a quick example of the pct_change function.

myles committed:
:bulb: Add some see also topics to pct_change.

myles committed:
:bulb: Stop showing the private method.

myles committed:
:bulb: Add an exmaple of Series and a Series with a fill_method.

myles committed:
:rotating_light: Fix a mistake with the docstring.

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         freq : DateOffset, timedelta, or offset alias string, optional
-            Increment to use from time series API (e.g. 'M' or BDay())
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        kwargs : mapping, optional
```
`kwargs : mapping, optional` -> `**kwargs`

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
-            Increment to use from time series API (e.g. 'M' or BDay())
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        kwargs : mapping, optional
+            A dictionary of keyword arguments passed into
```
It's not a dictionary, but just additional keywords, so you cna make this ""Additional keyword arguments are passed .."

jorisvandenbossche replied:
But on second thought, are there actually additional keywords for shift?

myles replied:
That's what it's mapped to https://github.com/pandas-dev/pandas/blob/7169830040255c1073f99e2ee8817596a67b1cd5/pandas/core/generic.py#L7561-L7562

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         Returns
         -------
-        chg : %(klass)s
+        chg : Series or DataFrame, same type as the input
```
'input' -> 'calling object'

jorisvandenbossche commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
+        See Also
+        --------
+        pandas.DataFrame.diff : see the difference of two columns
+        pandas.Series.diff : see the difference of two columns
```
Can you move this above the "Examples" section?

TomAugspurger replied:
Can also remove the `pandas.` prefix I think.

jorisvandenbossche reviewed (commented):
Nice PR! Added a few comments

myles committed:
:bulb: Update based off feedback from @jorisvandenbossche.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         freq : DateOffset, timedelta, or offset alias string, optional
-            Increment to use from time series API (e.g. 'M' or BDay())
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        **kwargs : mapping, optional
```
Can remove the type here. Just `**kwargs`.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
+            Increment to use from time series API (e.g. 'M' or BDay()).
+        **kwargs : mapping, optional
+            Additional keyword arguments are passed into
+            ``DataFrame.shift``/``Series.shift``.
```
Single backticks. "or" instead of `/`

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         Returns
         -------
-        chg : %(klass)s
+        chg : Series or DataFrame, same type as the calling object
```
```
change : Series or DataFrame
    The same type as the calling object.
```

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         Notes
         -----
-
         By default, the percentage change is calculated along the stat
```
I don't think most users will understand `stat` access. Best to remove this and only focus on Series / DataFrame.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
         axis: 0, or ``Index``, for ``DataFrame`` and 1, or ``minor`` for
         ``Panel``. You can change this with the ``axis`` keyword argument.
+
+        Percentage change in French franc, Deutsche Mark, and Italian lira from
```
Move under the "Examples" header.

TomAugspurger commented on pandas/core/generic.py:
```diff
@@ -7515,29 +7515,91 @@ def _check_percentile(self, q):
+        1980-01-01       NaN       NaN       NaN
+        1980-02-01  0.013810  0.013684  0.006549
+        1980-03-01  0.053365  0.059318  0.061876
+
```
Show an example with `axis='columns'`. Can use the same DataFrame, even though it's a silly example for axis=1.

myles committed:
:bulb: Update based off feedback from @TomAugspurger.

TomAugspurger committed:
Previous to prior

Note that it's immedate, but configurable.

no type on kwargs.

Example with periods.

Moved See Also.

TomAugspurger commented:
Made some touchups. Thanks @myles!

jorisvandenbossche commented:
@TomAugspurger small comment on when merging: can you remove all the different commit messages (in case of those DOC PRs it's mostly useless to keep them). It's maybe not that important and only my nitpicky self, but it's only a "click in box, ctrl-A, del" combo away :-)

TomAugspurger commented:
👍
